src/Microsoft.Extensions.Configuration.AzureAppConfiguration/OfflineFileCache.cs

```diff
@@ -170,20 +143,35 @@ public string Import(AzureAppConfigurationOptions options)
         /// <summary>
         /// An implementation of <see cref="IOfflineCache.Export(AzureAppConfigurationOptions, string)"/> that caches the data in the file system.
         /// </summary>
-        public void Export(AzureAppConfigurationOptions options, string data)
+        public void Export(AzureAppConfigurationOptions appConfigOptions, string data)
         {
-            EnsureOptions(options);
+            if (appConfigOptions == null)
+            {
+                throw new ArgumentNullException(nameof(AzureAppConfigurationOptions));
+            }
+
+            EnsureOptions(appConfigOptions);
+
+            DateTimeOffset expiryTime;
+            string expiryString;
+
+            try
+            {
+                expiryTime = DateTimeOffset.UtcNow + _options.FileCacheExpiration;
+                expiryString = expiryTime.ToString(CultureInfo.InvariantCulture);
+            }
+            catch (ArgumentOutOfRangeException)
+            {
+                throw new ArgumentOutOfRangeException($"{nameof(OfflineFileCacheOptions)}.{nameof(OfflineFileCacheOptions.FileCacheExpiration)}", "Please provide a shorter file cache expiration.");

```

Comment:
>ArgumentOutOfRangeException [](http://example.com/codeflow?start=26&length=27)

Is it reasonable to just set the expirty to max datetime here?

Reply:
Although its possible to set expiration to `DateTimeOffset.MaxValue`, we shouldn't be doing it because its discouraged in Data Protection docs. I think its safer to throw the exception instead of allowing users to create cache files that persist indefinitely.

Reply:
And that's even though we already validated the expiration when we validated options on construction? In this case it would have to be that the initial expiry was valid, but now it is not.

Reply:
I removed that check from options construction because we have to go through the same check here again.